viser/client/src/ControlPanel/BottomPanel.tsx

```diff
@@ -0,0 +1,213 @@
+import { Box, Paper } from "@mantine/core";
+import { IconCaretUp } from "@tabler/icons-react";
+import React from "react";
+
+const BottomPanelRefContext =
+  React.createContext<React.RefObject<HTMLDivElement> | null>(null);
+
+const pagePercent = (start: number, end: number) => { return (start - end) / window.innerHeight }
+const getHeight = (element: HTMLElement) => {
+  let size = 0;
+  const children = element.children
+  for (let i = 0; i < children.length; i++) {
+    size += children[i].clientHeight;
+  }
+  return size;
+}
+
+export default function BottomPanel({
+  children,
+}: {
+  children: string | React.ReactNode;
+}) {
+  const panelWrapperRef = React.useRef<HTMLDivElement>(null);
+  return (
+    <BottomPanelRefContext.Provider value={panelWrapperRef}>
+      <Paper
+        radius="0"
+        withBorder
+        sx={{
+          boxSizing: "border-box",
+          width: "100%",
+          zIndex: 100,
+          position: "fixed",
+          bottom: 0,
+          left: 0,
+          margin: 0,
+          overflow: "hidden",
+          minHeight: "3.5em",
+          height: "3.5em",
+          transition: "height 0.3s linear",
+          "& .expand-icon": {
+            transform: "rotate(0)",
+          },
+          "&.hidden .expand-icon": {
+            transform: "rotate(180deg)",
```

Comment:
the icon currently feels upside down to me? since it's now a "drag up" instead of a "drop-down"